lessons/9-intermediate-r-2/slides.qmd

```diff
@@ -12,73 +12,96 @@ format:
     footer: "[Reproduciblity & Data Science in R](../../index.html)"
 ---
 
-# Before we jump in...
+# Learning objectives
 
-## Show and tell (Late October)
+-   Write a more advanced function
+-   Set up an analysis with function and analysis scripts
+-   Use *iteration* to streamline workflows involving functions
 
--   An opportunity to apply workshop skills to your own work.
--   You could apply different parts to different projects, or more integratively to one single project.
--   Optional, but encouraged!
+# When should I write my own function?
 
-## Some inspirations
+-   Standardized procedures done many times
+-   Discrete chunks of a workflow
+-   Code used by multiple scripts or at multiple points in a workflow
 
--   Turn the project folder for one of your current research project into a git/GitHub repo and/or R project
--   Modify an R script to use functions, conditionals, or best-practices for code formatting
--   Improve directory organization for project or your entire computer!
+# When should I leave code in a script?
 
-# Today's objectives
+-   Single-use, highly-customized code (e.g. a plot)
+-   Code that is changing a lot
+-   Code that *tells the story* of your analysis
 
--   Modify our data cleaning function to apply to different islands
--   Use iteration to apply this function to each island
--   Organize our workflow into "functions" and "running" scripts
--   Learn how to get help with more advanced topics
+# Questions to ask yourself
 
-# Part I: Generalizing a function
+-   Do I find myself typing the same code over and over?
+-   Am I copy-pasting and changing a single variable/character?
+-   Does this code distract from the readability of a script?
+-   Is there an existing function I could pull from an R package?
 
--   Open up your Rproject from Tuesday
--   Coding objectives:
-    -   Make sure the function we wrote works
-    -   Apply this function to different islands by swapping URLS
-    -   Add "ifelse" logic to avoid having to re-type the URL every time
+# Coding time
 
-# Part II: Iteration
+-   Review sample analysis to decide what could be a function
+-   Organize this analysis into analysis and function scripts
 
-## Iteration by hand
+# Iteration
 
-``` r
-my_first_site <- read.csv("data_first.csv")
-my_second_site <- read.csv("data_second.csv")
-my_third_site <- read.csv("data_third.csv")
+# Iteration
```

Comment:
This slide's content is off the bottom of the screen when I preview it.  Use level 2 headers?